src/Core/Utilities/JsonHelpers.cs

```diff
@@ -129,4 +130,63 @@ public override void Write(Utf8JsonWriter writer, DateTime? value, JsonSerialize
             writer.WriteStringValue(CoreHelpers.ToEpocMilliseconds(value.Value).ToString());
         }
     }
+
+    /// <summary>
+    /// Allows reading a string from a number or string, should only be used on string properties
+    /// </summary>
+    public class PermissiveStringConverter : JsonConverter<string>
+    {
+        internal static PermissiveStringConverter Instance = new();
+
+        public override string Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            return reader.TokenType switch
+            {
+                JsonTokenType.String => reader.GetString(),
+                JsonTokenType.Number => reader.GetInt64().ToString(),
+                JsonTokenType.Null => null,
+                _ => throw new JsonException($"Unsupported TokenType: {reader.TokenType}"),
+            };
+        }
+
+        public override void Write(Utf8JsonWriter writer, string value, JsonSerializerOptions options)
+        {
+            writer.WriteStringValue(value);
+        }
+    }
+
+    /// <summary>
+    /// Allows reading a string from a number or string, should only be used on string properties
+    /// </summary>
+    public class PermissiveStringEnumerableConverter : JsonConverter<IEnumerable<string>>
+    {
+        public override IEnumerable<string> Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if (reader.TokenType != JsonTokenType.StartArray)
+            {
+                throw new JsonException("Should only be used on a JSON Array");
```

Comment:
For flexibility, please consider moving this line below the construction of `stringList` below, then test the token type similar to `PermissiveStringConverter`'s implementation (or use it's existing implementation in an internal status method for sharing that logic) so you can return the single item in an `IEnumerable<string>` instead of throwing an error; that way if JSON structure changes where some systems will use a single property in place of an array if there's only 1 item (I know, it's dumb but I've seen it too many times to discount the possibility).

Also, what happens when the token type is `Null`, which I don't see handled here either? Please also add that, or return an empty `IEnumerable<string>` in place of `null`; whatever is appropriate (I think most references to `IEnumerable<T>` expect a value (not `null`).